### DISCUSSION TITLE
How can I implement WalletConnect with Ethers 6

### DISCUSSION THREAD

**Discussion by petritnuredini** [WalletConnect/walletconnect-monorepo#2316] [Developer Support]
Hi everyone, recently I am working on a DAPP and need to implement the WalletConnect.
The problem is I am using Ethers 6.2.1 and as I understood the WalletConnect doesn't support the Ethers6.

Is there any work around or anything that might help me?

Thank you in advance ✌

**Comment by finessevanes** [comment]
Thanks for the question - forwarding to our team to see what options we have for you!

**Reply by finessevanes** [reply]
A workaround for now can be for you to use `web3modal/standalone`: https://docs.walletconnect.com/2.0/web3modal/advanced/standalone/sign/installation

Packages like `web3modal/react` and `web3modal/html` depend on wagmi, and since wagmi doesn't support ethers 6 at this moment, our packages won't be able to either.

**Reply by jxom** [reply]
wagmi will not be supporting ethers v6.

**Reply by paouvrard** [reply]
Now that `wagmi` has replaced `ethers` with `viem`, is there any way to use Web3Modal with ethers and injected wallets ? maybe with `web3modal/standalone` ?

**Reply by paouvrard** [reply]
This worked great for using an ethers Web3Provider with the latest Web3Modal and viem: https://wagmi.sh/react/ethers-adapters#wallet-client--signer
